File: rb/features.rb
# OpenMeteo SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module OpenMeteoFeatures
  def self.make_feature(name)
    case name
    when "base"
      OpenMeteoBaseFeature.new
    when "test"
      OpenMeteoTestFeature.new
    else
      OpenMeteoBaseFeature.new
    end
  end
end

File: py/features.py
# OpenMeteo SDK feature factory

from feature.base_feature import OpenMeteoBaseFeature
from feature.test_feature import OpenMeteoTestFeature


def _make_feature(name):
    features = {
        "base": lambda: OpenMeteoBaseFeature(),
        "test": lambda: OpenMeteoTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
